common/scala/src/main/scala/org/apache/openwhisk/core/database/mongodb/MongoDBArtifactStoreProvider.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.openwhisk.core.database.mongodb
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import org.apache.openwhisk.common.Logging
+import org.apache.openwhisk.core.ConfigKeys
+import org.apache.openwhisk.core.database._
+import org.apache.openwhisk.core.entity.size._
+import org.apache.openwhisk.core.entity.{DocumentReader, WhiskActivation, WhiskAuth, WhiskEntity}
+import org.mongodb.scala.MongoClient
+import pureconfig._
+import pureconfig.generic.auto._
+import spray.json.RootJsonFormat
+
+import scala.reflect.ClassTag
+
+case class MongoDBConfig(uri: String, database: String) {
+  assume(Set(database, uri).forall(_.nonEmpty), "At least one expected property is missing")
```

Comment:
hm, do you mean one parameter is an empty string `""` or `"          "`

Reply:
I mean, `Set(database, uri)` would be always nonEmpty.

Reply:
the `_.nonEmpty` will check against `database` and `url` in the `Set` instead of `Set` itself since there is a `forall`

```
scala> Set("", "asd").forall(_.nonEmpty)
res0: Boolean = false
```

Reply:
What I meant is this.

```
scala> Set("", "").forall(_.nonEmpty)
res0: Boolean = false
```

Reply:
so in this case `Set("", "")`, `assume` will raise an error, this is expected

Reply:
ok now I get it.